docs/source/external-structures.rst

```diff
@@ -43,6 +43,32 @@ A requirement for the user-specified containers is to provide a minimal ``std::v
 provides the aforementioned interface. In fact, the container could even be backed by a
 file on the disk, a database or a binary message.
 
+Adapting a pointer
+------------------
+
+Suppose that you want to use the *xtensor* machinery on a small contiguous subset of a large tensor.
+You can, of course, use :ref:`Views`, but for efficiency you can also use pointers to the right bit of memory.
+Consider an example of an ``[M, 2, 2]`` tensor ``A``,
+for which you want to operate on ``A[i, :, :]`` for different ``i``.
+In this case the most efficient *xtensor* has to offer is:
+
+.. code-block:: cpp
+
+    int main()
+    {
+        size_t M = 3;
+        xt::xarray<int> A = xt::arange<int>(M * 2 * 2).reshape({M, 2, 2});
+        auto b = xt::adapt(&A.flat(0), std::array<size_t, 2>{2, 2});
+
+        for (size_t i = 0; i < M; ++i) {
+            b.storage().replace(&A.flat(i * 2 * 2));
```

Comment:
- The storage should not be part of the public API for the end user, it is provided to allow optimizations in the implementation of xtensor. A `reset_buffer` or `reset_data` method should be provided in the adaptor classes (and that would call the `replace` method of the buffers. Of course this method should be available for adaptors on 1D buffers only, therefore the implementation should use SFINAE.
- `replace` could be renamed to be more expressive: `reset_data` or `reset_buffer`

Reply:
Do you mean, keep the renamed methods of this PR, and add a public API to the adaptors?